### DISCUSSION TITLE
How to stop the process from closing in NodeJS

### DISCUSSION THREAD

**Discussion by ghost** [telegraf/telegraf#2024] [Q&A]
**The Scenario is Broadcasting in Private Chats**

> What I have is an array of users obtained from MongoDB and I am using a forEach loop to send each of them a broadcast message

> ***Ignoring the fact that doing this broadcast forEach Loop without having a setInterval function will cause a 429 Error and will stop the process***

The Code: `bot.telegram.sendMessage('chatId', 'Hello This is The Bot');`

Now If the user hasn't blocked the bot, He will recieve a message from the bot as usual 

But If the user has blocked the bot, Telegraf will throw an error 403

The Error Code: ```
TelegramError: 403: Forbidden: bot was blocked by the user
    at Telegram.callApi (/home/runner/tgbotbhagwad/TelegramBotBhagwadGita/node_modules/telegraf/lib/core/network/client.js:315:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  response: {
    ok: false,
    error_code: 403,
    description: 'Forbidden: bot was blocked by the user'
  },
  on: {
    method: 'sendMessage',
    payload: { chat_id: 1234567890, text: 'hello' }
  }
}
```

> I would not ask a question here, but the thing is Try Catch wont work on this thing, So I am kinda forced to


Hoping a solution for this Thanks

**Comment by suprunchuk** [comment]
try this

```js
const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));

async function broadcastMessage(users, message) {
    for (const user of users) {
        try {
            await bot.telegram.sendMessage(user.chatId, message);
        } catch (error) {
            if (error.code === 403) {
                console.error(`User ${user.chatId} has blocked the bot.`);
            } else {
                console.error(`Failed to send message to user ${user.chatId}:`, error);
            }
        }
        // Implementa un retraso para evitar el error 429
        await delay(1000); // Espera 1 segundo entre cada mensaje
    }
}

// Ejemplo de uso:
const users = await getUsersFromMongoDB(); // Tu función para obtener los usuarios
const message = "Hello, this is the Bot!";
broadcastMessage(users, message);

```

**Comment by ghost** [comment]
Thank You Monsieur, I will Implement this

Appreciate You Brother
